docs/core/plugins/sieve.md

```diff
@@ -46,128 +58,191 @@ protocol lmtp {
+than a single `personal` storage can be defined, only the first one listed in
+the configuration is used. 
 
-Points to the directory where the compiled binaries for this script
-location are stored. This directory is created automatically if
-possible.
+The LDA Sieve plugin uses the personal storage to find the active script for
+Sieve filtering at delivery. If the storage supports storing more than a single
+script, personal scripts can also be retrieved by name. The Sieve include
+extension will then use this storage for retrieving `:personal` scripts and the
+ManageSieve service will be able to store the user's scripts there.
 
-If this option is omitted, the behavior depends on the location type.
+#### `after`
 
-Don't specify the same directory for different script locations, as this
-will result in undefined behavior. Multiple mail users can share a single
-script directory if the script location is the same and all users share
-the same system credentials (uid, gid).
+An `after` storage is the source of one script or several scripts that are to be
+executed after the user's personal script. If the storage supports storing more
+than a single script, these scripts will be executed in a well-defined order
+defined by the storage driver. Multiple `after` storages can be configured and
+each storage will be accessed in sequence to retrieve scripts for execution
+after the personal script. The storages will be accessed in the order these
+storages are defined in the configuration, unless the order is overridden by the
+[[setting,sieve_script_precedence]] setting.
+
+This is usually a global script, so be sure to pre-compile the specified
+script manually in that case using the sievec command line tool, as
+explained by [[man,sievec]].`
+
+#### `before`
+
+A `before` storage behaves identical to an `after` storage, except the contained
+script or scripts are run **before** user's personal script (instead of
+**after**).
+
+#### `default`
+
+The `default` storage yields the sieve script that gets executed **only** if
+the user's personal Sieve script does not exist. Although more
+than a single `default` storage can be defined, only the first one listed in
+the configuration is used. 
+
+If [[setting,sieve_script_name]] is set for this script storage, the default
+script can be seen and accessed by this name through ManageSieve (and
+doveadm sieve). See below ([[link,sieve_visible_default_script]]).
+
+This is usually a global script, so be sure to pre-compile the specified
+script manually in that case using the sievec command line tool, as
+explained by [[man,sievec]].`
+
+#### `discard`
+
+The `discard` storage yields the sieve script that gets executed for any message
+that is about to be discarded; i.e., it is not delivered anywhere by the normal
+Sieve execution. Although more than a single `discard` storage can be defined,
+only the first one listed in the configuration is used. The `discard` storage
```

Comment:
same, error?

Reply:
I am not deeply invested in this. Make a ticket if you want this changed.